src/dex/hashflow/hashflow.ts

```diff
@@ -0,0 +1,689 @@
+    } else if (!rfq.gasEstimate) {
+      const message = `${
+        this.dexKey
+      }: Failed to fetch RFQ for ${this.getPairName(
+        normalizedSrcToken.address,
+        normalizedDestToken.address,
+      )}. No gas estimate.`;
+      this.logger.warn(message);
+      throw new RfqError(message);
+    } else if (rfq.quoteData.rfqType !== RFQType.RFQT) {
+      const message = `${
+        this.dexKey
+      }: Failed to fetch RFQ for ${this.getPairName(
+        normalizedSrcToken.address,
+        normalizedDestToken.address,
+      )}. Invalid RFQ type.`;
+      this.logger.warn(message);
+      throw new RfqError(message);
+    }
+
+    assert(
+      rfq.quoteData.baseToken === normalizedSrcToken.address,
+      `QuoteData baseToken=${rfq.quoteData.baseToken} is different from srcToken=${normalizedSrcToken.address}`,
+    );
+    assert(
+      rfq.quoteData.quoteToken === normalizedDestToken.address,
+      `QuoteData baseToken=${rfq.quoteData.quoteToken} is different from srcToken=${normalizedDestToken.address}`,
+    );
+
+    return [
+      {
+        ...optimalSwapExchange,
+        data: {
+          mm: optimalSwapExchange.data!.mm,
+          quoteData: rfq.quoteData,
+          signature: rfq.signature,
+          gasEstimate: rfq.gasEstimate,
+        },
+      },
+      { deadline: BigInt(rfq.quoteData.quoteExpiry) },
+    ];
+  }
+
+  getCalldataGasCost(poolPrices: PoolPrices<HashflowData>): number | number[] {
+    return CALLDATA_GAS_COST.DEX_NO_PAYLOAD;
+  }
+
+  getTokenFromAddress?(address: Address): Token {
+    // We don't have predefined set of tokens with decimals
+    // Anyway we don't use decimals, so it is fine to do this
+    return { address, decimals: 0 };
+  }
+
+  getAdapterParam(
+    srcToken: string,
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: HashflowData,
+    side: SwapSide,
+  ): AdapterExchangeParam {
+    const { quoteData, signature } = data;
+
+    assert(quoteData !== undefined, `${this.dexKey}: quoteData undefined`);
+
+    const payload = this.routerInterface._abiCoder.encode(
+      [
+        'tuple(address pool, address quoteToken, address externalAccount, uint256 baseTokenAmount, uint256 quoteTokenAmount, uint256 quoteExpiry, uint256 nonce, bytes32 txid, bytes signature)',
+      ],
+      [
+        {
+          pool: quoteData.pool,
+          quoteToken: quoteData.quoteToken,
+          externalAccount: quoteData.eoa ?? ZERO_ADDRESS,
+          baseTokenAmount: quoteData.baseTokenAmount,
+          quoteTokenAmount: quoteData.quoteTokenAmount,
+          quoteExpiry: quoteData.quoteExpiry,
+          nonce: quoteData.nonce ?? 0,
+          txid: quoteData.txid,
+          signature,
+        },
+      ],
+    );
+
+    return {
+      targetExchange: this.routerAddress,
+      payload,
+      networkFee: '0',
+    };
+  }
+
+  getBlackListKey(address: Address) {
+    return `blacklist_${address}`.toLowerCase();
```

Comment:
better to namespace with dexKey [minor]

Reply:
It is already namespaced by dexKey and network in [func](https://github.com/paraswap/paraswap-dex-lib/blob/dd2493f694c7ca5489519a50072fa4943ff815d5/src/dex-helper/icache.ts#L12)